expo/expo
# [home] Remove explore and change profile pages to be account-based

# Why

The new account model on the backend is that all apps and snacks are owned by accounts (personal or organization). This didn't cause any breaking changes in home since it was querying them via the user which defaulted to their personal account. Unfortunately, this broke slightly when the user had access to other organization accounts as they were also included in the list, and when they were interacted with it would try to query them through the user query when they weren't associated with a user. Going forward, we want the home app information architecture to more closely resemble that of the website and therefore that of the backend.

This PR changes things fairly significantly:
- Remove all apps/snacks queries through the user object, and instead use the account object. This comes with a change to permissioning since only account members can see accounts and their objects.
- This means that any public profile pages we used to show in the app either a) need a new type of query that bypasses just lists an account's public apps/snacks, or b) need to be removed. Talking to @tcdavis, we opted to remove them. The instances where this was possible and that were removed were:
    - Explore page. This was only shown on android and looking at the usage metrics of the pre-filled projects revealed that it's not used very frequently.
    - Tapping on a username on the projects page (homepage).
    
Talking to @tcdavis it sounds like this is probably the direction that we want to take, moving away from being a public projects browser and towards being a browser for projects that you've been granted access to. Note that this doesn't change anything about what projects you can open, just which ones you can browse. In a follow-up, I'll add an interface to list all the organizations you are a part of so that all projects are accessible.

# How

1. Change all User profile screens to use Account queries instead
2. Remove instances of pages that would throw due to violated account permissions (public project browsing).

# Test Plan

On both iOS and Android:
1. Log in as a user with both snacks and apps in their personal account, ensure profile project and snack browsing still works.
2. Log out, ensure logged-out UI still works.
3. Log in as a user with no snacks or apps in their personal account, ensure profile empty state still works.
